server/src/main/java/org/opensearch/index/engine/NRTReplicationEngine.java

```diff
@@ -368,6 +368,11 @@ public boolean shouldPeriodicallyFlush() {
     @Override
     public void flush(boolean force, boolean waitIfOngoing) throws EngineException {
         ensureOpen();
+        // Skip flushing for indices with partial locality (warm indices)
+        // For these indices, we don't need to commit as we will sync from the remote store on re-open
+        if (engineConfig.getIndexSettings().isStoreLocalityPartial() == false) {
```

Comment:
shouldn't be this `engineConfig.getIndexSettings().isStoreLocalityPartial() `   ?